FullStack-DevTool/FSDT
# feat: :sparkles: create sdk package

Create the sdk package. 

### How to test? 

First of all, you need to build the packages. To do so, in packages/app run the command `yarn package` and in packages/sdk, run `yarn build`.

I suggest you to init a new project and then install the sdk and the fsdt app in this new project (using yarn link). Then, you should create 2 files : one for the monitor and the second one for the source. Finally, you can instantiate the FsdtLogger in both files : the source one should send the log and the monitor one should receive the log.

Obviously, the monitor will receive the logs as soon as the aggregator server is started.
You can start it by running this command `yarn fullstack-devtool -p 1234`

**Source file:**
```typescript
import FsdtLogger from "@fullstack-devtool/sdk";

const logger = new FsdtLogger("test-app-1", {
	port: 1234,
	connectionType: "source",
});

setInterval(() => {
	logger.info({ myVar: "myVar" });
}, 1000);
``` 

**Monitor file:**
```typescript
import FsdtLogger from "@fullstack-devtool/sdk";

const logger = new FsdtLogger("test-app-1", {
	port: 1234,
	connectionType: "monitor",
});

logger.onLogReceived((message) => {
	console.log({ message });
});

setInterval(() => {}, 1 << 30);

```
